static/src/javascripts/projects/common/modules/experiments/tests/recommended-for-you.js

```diff
@@ -76,13 +76,34 @@ define([
             if (recommendations && new Date(recommendations.expiry) > new Date()) {
                 insertSection(recommendations.items);
             } else {
-                var promisedRecommendations = getRemoteRecommendations();
+                var promisedRecommendations = getRemoteRecommendationsIds().then(getCardsHtml);
                 promisedRecommendations.then(cacheRecommendations);
                 promisedRecommendations.then(insertSection);
             }
         }
 
-        function getRemoteRecommendations() {
+        function getCardsHtml(items) {
+            return Promise.all(items.map(getCardHtml));
+        }
+
+        function getCardHtml(id) {
+            var endpoint = '/embed/contentcard/' + id + '.json';
+            var request = fetch(endpoint, {
```

Comment:
There's a convenience [`fetchJson`](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/utils/fetch-json.js) module to handle the JSON stuff for you.